python/cudf/cudf/tests/test_csv.py

```diff
@@ -1532,51 +1547,38 @@ def test_csv_writer_datetime_data(tmpdir):
 @pytest.mark.parametrize(
     "index", [True, False, np.bool_(True), np.bool_(False)]
 )
-@pytest.mark.parametrize(
-    "line_terminator", ["\r", "\n", "NEWLINE", "<<<<<", np.str_("\n\r")]
-)
-def test_csv_writer_mixed_data(
-    sep, columns, header, index, line_terminator, tmpdir
-):
-    pdf_df_fname = tmpdir.join("pdf_df_3.csv")
-    gdf_df_fname = tmpdir.join("gdf_df_3.csv")
+def test_csv_writer_column_and_header_options(columns, header, index):
+    pdf = make_numpy_mixed_dataframe()
+    df = cudf.from_pandas(pdf)
 
-    df = make_numpy_mixed_dataframe()
-    df["Date"] = df["Date"].astype("datetime64")
-    gdf = cudf.from_pandas(df)
-    gdf["Date"] = gdf["Date"].astype("datetime64[s]")
-    df.to_csv(
-        path_or_buf=pdf_df_fname,
-        index=index,
-        sep=sep,
-        columns=columns,
-        header=header,
-        line_terminator=line_terminator,
-        date_format="%Y-%m-%dT%H:%M:%SZ",
-        quoting=csv.QUOTE_NONE,
-        escapechar="\\",
-    )
-    gdf.to_csv(
-        path_or_buf=gdf_df_fname,
-        index=index,
-        sep=sep,
-        columns=columns,
-        header=header,
-        line_terminator=line_terminator,
+    cudf_buffer = BytesIO()
+    df.to_csv(cudf_buffer, columns=columns, header=header, index=index)
+    pd_buffer = BytesIO()
+    pdf.to_csv(pd_buffer, columns=columns, header=header, index=index)
+
+    expected = cudf.read_csv(pd_buffer, header=0 if header else None)
+    got = cudf.read_csv(cudf_buffer, header=0 if header else None)
+
+    expected_column_cnt = (
+        (1 if index else 0) + len(columns) if columns else got.shape[1]
```

Comment:
Should `expected_column_cnt` be based off `got.shape[1]`, or should it be `expected.shape[1]`?

Reply:
Fixed to use the input dataframe size, thanks for pointing this out!